### PR TITLE
Upgrade to 2.13 + setup docker-mysql for tests

### PR DESCRIPTION
**Pull request by arvindvjkumar**
I didn't get all postgres tests working (the ones that do auth stuff), but we don't use that code so I don't really care.

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -84,8 +83,9 @@ object Configuration {
     ,
     testOptions in Test += Tests.Argument(TestFrameworks.Specs2, "sequential"),
     scalacOptions in doc := Seq("-doc-external-doc:scala=http://www.scala-lang.org/archives/downloads/distrib/files/nightly/docs/library/"),
-    crossScalaVersions := Seq(projectScalaVersion, "2.10.6", "2.11.8"),
-    javacOptions := Seq("-source", "1.6", "-target", "1.6", "-encoding", "UTF8"),
+    scalaVersion := "2.13.6",
```
Comment (bibachhetri) [inline project/Build.scala:86]: This is probably fine but @seanwentzel can you confirm? Right now, [CODEZ uses](https://github.com/Asana/codez/blob/3dd6c5c9b83b93339a006aea41853e15836d132f/asana2/third_party/bazel_workspace/scala_repositories.bzl#L254) `2.13.10` but I think we could change this to 2.13.6 easily. Is that right?

File: docker/db-async/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+services:
+  mysql:
+    image: mysql:5.7.41
```
Comment (bibachhetri) [inline docker/db-async/docker-compose.yml:3]: Is this an issue for us? We no longer use mysql 8 everywhere right?

**Review by bibachhetri — CHANGES_REQUESTED** [review]
left a few questions on stuff that I have context on but you should likely get a review for the scala changes themselves since I don't have enough context to do a good review

**Comment by seanwentzel** [comment]
Confused why we forked this vs using https://mvnrepository.com/artifact/com.github.postgresql-async/postgresql-async

File: project/Build.scala (hunk 1 above)
Comment (seanwentzel) [reply project/Build.scala:86]: should be ok but I'd much prefer if either we don't fork this or we merge it into codez

**Comment by arvindvjkumar** [comment]
> Confused why we forked this vs using https://mvnrepository.com/artifact/com.github.postgresql-async/postgresql-async

Ah I didn't see that before when I looked at possible project successors. The main reason we forked it was a perf issue Yannis found (https://app.asana.com/0/1200752962001810/1204810580793065/f). I figured that I might as well make it compatible with 2.13 at the same time. The patch for that bug is here (https://github.com/Asana/postgresql-async/pull/1/files)

I took a look at that project you linked (https://github.com/postgresql-async/postgresql-async) and it looked like it's more actively maintained, but still has the underlying bug. 

Why don't I?
- Close this change + this fork
- Switch to using that dep to unblock Scala 2.13 work
- Bump mypipe to 2.13
- Open a PR against that repo with a bit more background on the perf issue

**Comment by seanwentzel** [comment]
SG, although I'd prioritize the perf issue more highly than the upgrade to 2.13 if it is urgent

**Pull request closed without merging**